qemu/run.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+QEMU_PATH="/opt/ILLIXR/qemu/build/x86_64-softmmu"
+
+# Download ubuntu-18.04.5-desktop-amd64.iso if it doesn't exist already
+if [ ! -f ubuntu-18.04.5-desktop-amd64.iso ]; then
+    wget https://releases.ubuntu.com/18.04/ubuntu-18.04.5-desktop-amd64.iso
+fi
+
+# Create a qcow2 image, if one doesn't exist, and launch with ubuntu installed
+if [ ! -f illixr.qcow2 ]; then
```

Comment:
If you only check for building qcow2 in the if-else, the launch code (lines 14-20) won't have to be duplicated (lines 25-31) and can be written just once after the if-else.

Reply:
The second case is slightly different to the first in that it launches without the ubuntu ISO attached. That way we boot into the qcow and not back into the installer

Reply:
This way if the user later deletes the ubuntu iso there's no issues at all, everything just works (tm)

Reply:
Oh, interesting. I didn't know that `qemu-img create` carried over state to the launch command.

Sounds good!

Reply:
It doesn't carry over state, its just that if the image hasn't been created then we know the user needs to create an image and install ubuntu. Once the image is created we assume they've successfully installed ubuntu (otherwise, they should remove the image and re-run the script per the instructions)

Reply:
Its really "2 scripts in one"- an "install_ubuntu" and "run_ubuntu" just packaged into one that should just do the right thing

Reply:
Sorry, I don't know what I'm missing here. Is the following not logically equivalent to what is in the script at the moment?

```
if [ ! -f illixr.qcow2 ]; then
    qemu-img create -f qcow2 illixr.qcow2 30G
fi

# Launch illixr.qcow2
$QEMU_PATH/qemu-system-x86_64 -enable-kvm -M q35 -smp 2 -m 4G \
        -hda illixr.qcow2 \
        -net nic,model=virtio \
        -net user,hostfwd=tcp::2222-:22 \
        -vga virtio \
        -cdrom ubuntu-18.04.5-desktop-amd64.iso \
        -display sdl,gl=on
```

Reply:
Yes, the key difference is the `-cdrom` line- when we detect the existence of `illixr.qcow2` we run qemu without the cdrom inserted. That's the only difference but it's a significant one (this way we boot into the installed ubuntu and not the CD, and it also lets the user remove the installer media to save disk space).

Reply:
But the `-cdrom` line is the same in both branches, no?

```
IF:   -cdrom ubuntu-18.04.5-desktop-amd64.iso \
ELSE: -cdrom ubuntu-18.04.5-desktop-amd64.iso \
```

Reply:
Good catch- that was a typo on my part. I did some looking into it though and there was a bigger issue with the behavior after removing the installer media. I pushed a fix that should address both of these issues with the correct behaviour